Tests/SonarScanner.MSBuild.PreProcessor.Test/RoslynAnalyzerProviderTests.cs

```diff
@@ -230,147 +175,157 @@ private List<SonarRule> CreateRules() =>
-            """;
-        File.ReadAllText(ruleSetPath).Should().Be(expectedContent, "Ruleset file does not have the expected content: {0}", ruleSetPath);
-    }
+        public void AssertExpectedPluginsRequested(IEnumerable<string> plugins) =>
+            MockInstaller.AssertExpectedPluginsRequested(plugins);
 
-    private void CheckExpectedAdditionalFileExists(string expectedFileName, string expectedContent, AnalyzerSettings actualSettings)
-    {
-        // Check one file of the expected name exists
-        var matches = actualSettings.AdditionalFilePaths.Where(x => string.Equals(expectedFileName, Path.GetFileName(x), StringComparison.OrdinalIgnoreCase));
-        matches.Should().ContainSingle("Unexpected number of files named \"{0}\". One and only one expected", expectedFileName);
+        public void AssertExpectedAdditionalFileExists(string expectedContent, string expectedFileName = "SonarLint.xml")
+        {
+            var filePaths = ActualSettings.AdditionalFilePaths.Should().ContainSingle();
+            // Check one file of the expected name exists
+            var matches = ActualSettings.AdditionalFilePaths.Where(x => string.Equals(expectedFileName, Path.GetFileName(x), StringComparison.OrdinalIgnoreCase));
+            matches.Should().ContainSingle("Unexpected number of files named \"{0}\". One and only one expected", expectedFileName);
+            // Check the file exists and has the expected content
+            var actualFilePath = matches.First();
+            File.Exists(actualFilePath).Should().BeTrue("AdditionalFile does not exist: {0}", actualFilePath);
+            // Dump the contents to help with debugging
+            TestContext.AddResultFile(actualFilePath);
+            TestContext.WriteLine("File contents: {0}", actualFilePath);
+            TestContext.WriteLine(File.ReadAllText(actualFilePath));
+            TestContext.WriteLine(string.Empty);
+            if (expectedContent is not null)
+            {
+                File.ReadAllText(actualFilePath).Should().Be(expectedContent, "Additional file does not have the expected content: {0}", expectedFileName);
+            }
+        }
 
-        // Check the file exists and has the expected content
-        var actualFilePath = matches.First();
-        File.Exists(actualFilePath).Should().BeTrue("AdditionalFile does not exist: {0}", actualFilePath);
+        private void CheckRuleset(string ruleSetPath, bool isDeactivated)
+        {
+            ruleSetPath.Should().NotBeNullOrEmpty("Ruleset file path should be set");
+            Path.IsPathRooted(ruleSetPath).Should().BeTrue("Ruleset file path should be absolute");
+            File.Exists(ruleSetPath).Should().BeTrue("Specified ruleset file does not exist: {0}", ruleSetPath);
+            TestContext.AddResultFile(ruleSetPath);
+            CheckFileIsXml(ruleSetPath);
+            var expectedFileName = isDeactivated ? $"Sonar-{Language}-none.ruleset" : $"Sonar-{Language}.ruleset";
```

Comment:
This method is hard to follow. 
Put all `var expected...` at the top.
Put both `Path....` checks from the bottom after the first one.

Like this, the method will have logical structure and will be easider to follow (declare, validate basic stuff, add result file, validate content)